### PR TITLE
[Fix-14660] [Resource Center] Remove NONE storage type and set LOCAL type as default

### LINKED ISSUE #14660: [Bug] [Resource Center] Set NONE type resource will throw NPE after login
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Login failed, and here are backend error logs:
```
[ERROR] 2023-07-29 11:27:24.382 +0800 org.apache.dolphinscheduler.api.exceptions.ApiExceptionHandler:[52] - resource not exist
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.api.service.impl.ResourcesServiceImpl.queryResourceBaseDir(ResourcesServiceImpl.java:1712)
	at org.apache.dolphinscheduler.api.service.impl.ResourcesServiceImpl$$FastClassBySpringCGLIB$$d2d5d7f6.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy.invokeMethod(CglibAopProxy.java:386)
	at org.springframework.aop.framework.CglibAopProxy.access$000(CglibAopProxy.java:85)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:704)
	at org.apache.dolphinscheduler.api.service.impl.ResourcesServiceImpl$$EnhancerBySpringCGLIB$$22bef631.queryResourceBaseDir(<generated>)
	at org.apache.dolphinscheduler.api.controller.ResourcesController.queryResourceBaseDir(ResourcesController.java:776)
	at org.apache.dolphinscheduler.api.controller.ResourcesController$$FastClassBySpringCGLIB$$2270281e.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)

...
```

### What you expected to happen

No error.

### How to reproduce

Set resource type to `NONE` and login.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Radeity**
## Purpose of the pull request
- close #14660

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by ruanwenjun** [comment]
It's better to add a StoragePlugin type with None rather than do some specific logic?

**Comment by Radeity** [comment]
> It's better to add a StoragePlugin type with None rather than do some specific logic?

Sounds good to me, I will finish it later today.

**Comment by ruanwenjun** [comment]
> > It's better to add a StoragePlugin type with None rather than do some specific logic?
> 
> Sounds good to me, I will finish it later today.

Remove the NONE type, use local storage is another choice.

**Comment by Radeity** [comment]
> Remove the NONE type, use local storage is another choice.

You're right, we can remove NONE type, but still to have to keep this check to avoid NPE. Then, I only need to update doc.

Only one thing have to be considered, LOCAL type can write files on disk which behaves a little different from NONE type, WDYT?

**Comment by Radeity** [comment]
Hi @ruanwenjun , I thought again, and suggest keep current fix in this PR and do not remove NONE type. We have the check in `checkResourceUploadStartupState()` for resources CURD and can return message `STORAGE_NOT_STARTUP` when setting NONE type. It's more user-friendly, in addition, use LOCAL type is kind of confused on my side.

**Comment by ruanwenjun** [comment]
> Hi @ruanwenjun , I thought again, and suggest keep current fix in this PR and do not remove NONE type. We have the check in `checkResourceUploadStartupState()` for resources CURD and can return message `STORAGE_NOT_STARTUP` when setting NONE type. It's more user-friendly, in addition, use LOCAL type is kind of confused on my side.

So we can use DEFAULT as default, and open storage is must, we don't support use `NONE`, then we can remove all check storage startup logic.

**Comment by Radeity** [comment]
> So we can use DEFAULT as default, and open storage is must, we don't support use `NONE`, then we can remove all check storage startup logic.

Do you mean we can add a default StoragePlugin, doing nothing when handling CURD requests of resources (behave like `NONE` type, after all, some users do not need storage)?

**Comment by ruanwenjun** [comment]
> > So we can use DEFAULT as default, and open storage is must, we don't support use `NONE`, then we can remove all check storage startup logic.
> 
> Do you mean we can add a default StoragePlugin, doing nothing when handling CURD requests of resources (behave like `NONE` type, after all, some users do not need storage)?



Sorry, we can use Local as default.

File: docs/docs/en/guide/resource/configuration.md [hunk 1]
```diff
@@ -80,8 +80,8 @@ data.basedir.path=/tmp/dolphinscheduler
 # resource view suffixs
 #resource.view.suffixs=txt,log,sh,bat,conf,cfg,py,java,sql,xml,hql,properties,json,yml,yaml,ini,js
 
-# resource storage type: HDFS, S3, OSS, GCS, ABS, NONE
-resource.storage.type=NONE
+# resource storage type: LOCAL, HDFS, S3, OSS, GCS, ABS
+resource.storage.type=LOCAL
```
Comment (zhongjiajie) [inline docs/docs/en/guide/resource/configuration.md:84]: My bad, seems I forget to change to docs when I use `LOCAL` as default

**Review by zhongjiajie — DISMISSED** [review]
Thanks for correct this

File: docs/docs/en/guide/resource/configuration.md (hunk 1 above)
Comment (Radeity) [reply docs/docs/en/guide/resource/configuration.md:84]: Never mind, should always say thanks to everyone who help maintain the doc :D

File: dolphinscheduler-common/src/main/resources/common.properties [hunk 2]
```diff
@@ -21,7 +21,7 @@ data.basedir.path=/tmp/dolphinscheduler
 # resource view suffixs
 #resource.view.suffixs=txt,log,sh,bat,conf,cfg,py,java,sql,xml,hql,properties,json,yml,yaml,ini,js
 
-# resource storage type: LOCAL, HDFS, S3, OSS, GCS, ABS, NONE. LOCAL type is a specific type of HDFS with "resource.hdfs.fs.defaultFS = file:///" configuration
```
Comment (Gallardot) [inline dolphinscheduler-common/src/main/resources/common.properties]: Why was NONE removed? Is storage required even if I'm just using K8S tasks?

Comment (Radeity) [reply dolphinscheduler-common/src/main/resources/common.properties]: We will use `LOCAL` type by default, you can see the above discussion. In this way, we can remove null storage plugin check for resource CURD api. In your scenario, you can ignore this configuration.

**Comment by Radeity** [comment]
Hi @zhongjiajie , UT test always fail, I've checked error log and find this:
```
java.lang.IllegalArgumentException: java.net.UnknownHostException: mycluster
```
Init step of HDFSStorageOperator will fail because of this exception, would you like to help check the test environment?

File: dolphinscheduler-common/src/main/resources/common.properties (hunk 2 above)
Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/resources/common.properties]: Yes, if you don't need to use resource center, you don't need to take care of this configuration.

**Comment by zhongjiajie** [comment]
I restart all the ci, let check what happened next

**Comment by Radeity** [comment]
Would you like to help review this PR when you are free? @zhongjiajie

**Review by zhongjiajie — APPROVED** [review]
Looks good and thanks for your contributed

**Pull request merged**